source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -220,8 +222,9 @@ Wavm::~Wavm() {
     delete emscriptenInstance_;
   }
   context_ = nullptr;
-  envoyModuleInstance_ = nullptr;
-  envoy_functions_.clear();
+  intrinsicModuleInstances_.clear();
```

Comment:
Question:  why these 3 clean up are needed? IMHO those 3 collections will be clean uponce the ~Wavm() completes